ddl/generated_column.go

```diff
@@ -50,10 +50,28 @@ func verifyColumnGeneration(colName2Generation map[string]columnGenerationInDDL,
 	return nil
 }
 
+// verifyColumnGenerationSingle is for ADD GENERATED COLUMN, we just need verify one column itself.
+func verifyColumnGenerationSingle(dependCols map[string]struct{}, cols []*table.Column, position *ast.ColumnPosition) error {
```

Comment:
The first parameter of this function is `dependCols`, but when call this function, you passed the parameter named `duplicateColNames`, is them same?
https://github.com/pingcap/tidb/blob/a7cf2762510cb1f4efd14d6ddd1ae08558ded4e8/ddl/ddl_api.go#L2113

Reply:
Same. Since `dependCols` will change in `checkDependedColExist`, so I make it copy.